src/ros/ros.ts

```diff
@@ -64,4 +71,29 @@ export interface ROSApi {
     activateRoslaunch: (launchFilepath: string, argument: string) => vscode.Terminal;
 }
 
-export let rosApi: ROSApi = new ros1.ROS1();
+const ros1Api: ROSApi = new ros1.ROS1();
+const ros2Api: ROSApi = new ros2.ROS2();
+const unknownRosApi: ROSApi = new unknownROS.UnknownROS();
+
+export let rosApi: ROSApi = unknownRosApi;
+
+export function selectROSApi(distro: string) {
+    rosApi = unknownRosApi;
+    switch(distro.toLowerCase()) {

```

Comment:
since the lists might keep updating, just wondering if it'd be better to have 2 arrays to store ROS distro names: `ros1Distros = ["kinetic", ...]`, and `ros2Distros = ["ardent", ...]`, so we can use [array.includes](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes) or [array.indexOf](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/indexOf)
```typescript
if (ros1Distros.indexOf(distro.toLowerCase()) !== -1) {
    // ...
}
if (ros2Distros.indexOf(distro.toLowerCase()) !== -1) {
    // ...
}
```

Reply:
I found the REP for ROS_VERSION usage, so I swap to use that to check the ROS1 & ROS2.